kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/http/StandardHttpClient.java

```diff
@@ -108,6 +108,13 @@ private CompletableFuture<HttpResponse<AsyncBody>> consumeBytesOnce(StandardHttp
     final Consumer<List<ByteBuffer>> effectiveConsumer = consumer;
 
     CompletableFuture<HttpResponse<AsyncBody>> cf = consumeBytesDirect(effectiveRequest, effectiveConsumer);
+    cf.exceptionally(throwable -> {
+      builder.interceptors.forEach((s, interceptor) -> interceptor.afterConnectionFailure(effectiveRequest, throwable));
+      if (throwable instanceof CompletionException) {
+        throw (CompletionException) throwable;
+      }
```

Comment:
sonar is complaining this branch is untested, but I can't work out a way to inject a `CompletionException`.  Any pointers gratefully received.

Reply:
Came up with https://github.com/SamBarker/kubernetes-client/commit/43f88fab5597a3d85bfedda2dab1033259f238de  which does trigger the branch using the Jetty client.

Reply:
apparently not according to sonar. It seemed to work in the debugger locally.

Reply:
Urgh yeah I hadn't thought about the multiple modules pushing in different directions. Sonar only reports a single number on the `StandardHttpClient` so a bit hard to know. I suspect on balance its probably better to have the test than not but I only added it to try and pass the coverage stats for the PR so not wedded to it as the core flows are tested.